File: Backend/carbonfootprint/src/main/java/com/carbonfootprint/carbonfootprint/service/AdminSupportService.java
package com.carbonfootprint.carbonfootprint.service;

import com.carbonfootprint.carbonfootprint.dto.SupportResponse;

import java.util.List;

public interface AdminSupportService {

    List<SupportResponse> getAllTickets();

    SupportResponse getTicketById(Long id);

    SupportResponse replyToTicket(Long id, String reply);

    SupportResponse updateStatus(Long id, String status);

    void deleteTicket(Long id);

}

File: frontend/footprint/src/admin/services/adminSupportService.js
import axios from "axios";

const API = "http://localhost:8080/api/admin/support";

const getToken = () => ({
    headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
    }
});

export const getAllTickets = async () => {
    const response = await axios.get(API, getToken());
    return response.data;
};

export const getTicketById = async (id) => {
    const response = await axios.get(`${API}/${id}`, getToken());
    return response.data;
};

export const replyTicket = async (id, data) => {
    const response = await axios.put(`${API}/${id}/reply`, data, getToken());
    return response.data;
};

export const updateStatus = async (id, status) => {
    const response = await axios.put(
        `${API}/${id}/status`,
        { status },
        getToken()
    );
    return response.data;
};

export const deleteTicket = async (id) => {
    await axios.delete(`${API}/${id}`, getToken());
};